samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/TypeFacadeQueries.java

```diff
@@ -82,22 +82,17 @@ public TypeFacade getTypeFacadeById(Long typeId) {
      * @return org.osid.shared.Type or null if no types of that id
      */
     public Type getTypeById(Long typeId){
-    	//pbd hack
-    	List typeList = getAllTypes();
-    	this.typeFacadeMap = createTypeFacadeMapById(typeList);
-    	
-    	// end pbd hack
```

Comment:
@master-bob is pbd Paul? If yes, can you ask him what this hack was intended to do? I can't figure it out .... it seems like a pure performance killer